### PR TITLE
fix(app): includeFixitCommands in GET run commands

### PR DESCRIPTION
**Pull request by TamarZanzouri**
# Overview

closes [EXEC-704](https://opentrons.atlassian.net/browse/EXEC-704).
revert do not render fixit commands and add query string to req.

## Test Plan and Hands on Testing

- load an ER protocol.
- enter ER.
- dispatch a few fixit commands.
- finish run.
- make sure there are no fixit commands in the run preview. 

## Changelog

revert commend intent logic from UI.
add `includeFixitCommands` in query string to GET run commands.

## Review requests

am I missing anyhting? 

## Risk assessment

low. no change in appearance.

[EXEC-704]: https://opentrons.atlassian.net/browse/EXEC-704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mjhuff — APPROVED** [review]
Looks great! Thank you for fixing this!

**Comment by TamarZanzouri** [comment]
tested and works as expected!

**Review by sfoster1 — APPROVED** [review]
Okay, so this is moving from client-side filtering to server-side filtering, right? Nice!

File: app/src/resources/runs/useNotifyAllCommandsQuery.ts [hunk 1]
```diff
@@ -22,7 +26,13 @@ export function useNotifyAllCommandsQuery<TError = Error>(
     options,
   })
 
-  const httpQueryResult = useAllCommandsQuery(runId, params, queryOptionsNotify)
+  const queryParams = params ?? null
```
Comment (mjhuff) [inline app/src/resources/runs/useNotifyAllCommandsQuery.ts]:
```suggestion
```
We shouldn't need this line.

File: api-client/src/runs/commands/getCommandsAsPreSerializedList.ts [hunk 2]
```diff
@@ -4,13 +4,13 @@ import type { ResponsePromise } from '../../request'
...
   CommandsAsPreSerializedListData,
-  GetCommandsParams,
+  GetRunCommandsParams,
 } from './types'
 
 export function getCommandsAsPreSerializedList(
   config: HostConfig,
   runId: string,
-  params: GetCommandsParams
+  params: GetRunCommandsParams
```
Comment (mjhuff) [inline api-client/src/runs/commands/getCommandsAsPreSerializedList.ts]: This should use `GetRunCommandsParamsRequest`

File: react-api-client/src/runs/useAllCommandsAsPreSerializedList.ts [hunk 3]
```diff
@@ -32,6 +32,11 @@ export function useAllCommandsAsPreSerializedList<TError = Error>(
     enabled: host !== null && runId != null && options.enabled !== false,
   }
   const { cursor, pageLength } = nullCheckedParams
+  const nullCheckedFixitCommands = params?.includeFixitCommands ?? undefined
```
Comment (mjhuff) [inline react-api-client/src/runs/useAllCommandsAsPreSerializedList.ts]:
```suggestion
  const nullCheckedFixitCommands = params?.includeFixitCommands ?? null
```

Comment (mjhuff) [reply react-api-client/src/runs/useAllCommandsAsPreSerializedList.ts]: Requires the type change to work

File: react-api-client/src/runs/useAllCommandsQuery.ts [hunk 4]
```diff
@@ -3,20 +3,22 @@ import { getCommands } from '@opentrons/api-client'
...
+export const DEFAULT_PARAMS: GetRunCommandsParamsRequest = {
   cursor: null,
   pageLength: DEFAULT_PAGE_LENGTH,
+  includeFixitCommands: null,
 }
 
 export function useAllCommandsQuery<TError = Error>(
   runId: string | null,
-  params?: GetCommandsParams | null,
+  params?: GetRunCommandsParams | null,
```
Comment (mjhuff) [inline react-api-client/src/runs/useAllCommandsQuery.ts]:
```suggestion
  params?: GetRunCommandsParamsRequest | null,
```

**Pull request merged**